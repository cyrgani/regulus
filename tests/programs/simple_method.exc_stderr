NameError: object has no method `nonexistent`
at programs/simple_method.re:19:33
at programs/simple_method.re:19:31

ArgumentError: expected `1` args, found `2` args for `<object>.double`

TypeError: 5 is not a Object!

ArgumentError: `@` expected the name of a method as second arg

