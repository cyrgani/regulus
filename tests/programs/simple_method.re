type(
    Foo,
    a,

    =(double, fn(self,
        Foo(*(.(self, a), 2))
    )),
),

=(b, Foo(3)),
=(d, .(b, double)),
=(c, d(b)),
assert_eq(.(c, a), 6),

=(x, Foo(5)),
=(y, @(x, double)),
assert_eq(.(y, a), 10),

__builtin_file_catch_assert_eq(@(x, nonexistent)),
__builtin_file_catch_assert_eq(@(x, double, 2)),
__builtin_file_catch_assert_eq(@(5, foo)),
__builtin_file_catch_assert_eq(@(x, fn(self, 2))),

import(aliases),
=(y, call_method(x, double)),
assert_eq(.(y, a), 10),