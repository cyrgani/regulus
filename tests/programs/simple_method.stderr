NameError: object has no method `nonexistent`
at programs/simple_method.re:19:24
at programs/simple_method.re:19:22
ArgumentError: expected `1` args, found `2` args for `<object>.double`
at programs/simple_method.re:20:24
at programs/simple_method.re:20:22
TypeError: 5 is not a Object!
ArgumentError: `@` expected the name of a method as second arg
